src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -250,6 +254,20 @@ namespace System.Diagnostics.Tracing
     /// </remarks>
     public partial class EventSource : IDisposable
     {
+
+#if FEATURE_EVENTSOURCE_XPLAT
+        private static XplatEventLogger persistent_Listener = null;
+
+        static EventSource()
+        {
+            if ((persistent_Listener == null) && XplatEventLogger.IsEventSourceLoggingEnabled())
+            {
+                persistent_Listener = new XplatEventLogger();
```

Comment:
I am not sure what you mean, can you point me to other instances of where we check for allocation failures ?